system/services/admin/DataManagerBatchOperationService.cfc

```diff
@@ -0,0 +1,429 @@
+/**
+ * Service to provide batch operation logic. i.e. logic for batch record
+ * operations kicked off from datamanager datatables.
+ *
+ * @singleton
+ * @presideservice
+ * @autodoc
+ */
+component displayName="Data manager batch operation service" {
+
+// CONSTRUCTOR
+	/**
+	 *
+	 * @customizationService.inject datamanagerCustomizationService
+	 */
+	public any function init(
+		required any customizationService
+	) {
+		_setCustomizationService( arguments.customizationService );
+
+		return this;
+	}
+
+// PUBLIC METHODS
+	public boolean function batchEditField(
+		  required string  objectName
+		, required string  fieldName
+		, required string  value
+		,          array   sourceIds          = []
+		,          boolean batchAll           = false
+		,          struct  batchSrcArgs       = {}
+		,          string  multiEditBehaviour = "append"
+		,          string  auditAction        = "datamanager_batch_edit_record"
+		,          string  auditCategory      = "datamanager"
+		,          string  auditUserId        = ""
+		,          any     logger
+		,          any     progress
+	) {
+		var pobjService       = $getPresideObjectService();
+		var isMultiValue      = pobjService.isManyToManyProperty( arguments.objectName, arguments.fieldName );
+		var canLog            = StructKeyExists( arguments, "logger" );
+		var canInfo           = canLog && arguments.logger.canInfo();
+		var canWarn           = canLog && arguments.logger.canWarn();
+		var canReportProgress = StructKeyExists( arguments, "progress" );
+		var totalrecords      = ArrayLen( sourceIds );
+		var hasLabelField     = Len( pobjService.getLabelField( arguments.objectName ) );
+		var processed         = 0;
+		var objectTitle       = "";
+		var fieldTitle        = "";
+		var uriRoot           = "";
+		var moreToFetch       = arguments.batchAll;
+		var queueId           = "";
+
+		if ( arguments.batchAll ) {
+			queueId      = queueBatchOperation( arguments.objectName, arguments.batchSrcArgs );
+			totalRecords = getBatchSourceRecordCount( arguments.objectName, arguments.batchSrcArgs );
+		}
+
+		if ( canInfo ) {
+			uriRoot = pobjService.getResourceBundleUriRoot( arguments.objectName );
+			objectTitle = $translateResource( uri=uriRoot & "title.singular", defaultValue=arguments.objectName );
+			fieldTitle  = $translateResource( uri=uriRoot & "field.#arguments.fieldName#.title", defaultValue=arguments.fieldName );
+			arguments.logger.info( $translateResource( uri="cms:datamanager.batchedit.task.starting.message", data=[ objectTitle, fieldTitle, NumberFormat( totalRecords ) ] ) );
+		}
+
+		try {
+			do {
+				if ( arguments.batchAll ) {
+					sourceIds = getNextBatchRecordsFromQueue( queueId, 100 );
```

Comment:
Slightly concerns me having 100 hard-coded in 2 places. Potentially someone could change the top one to 1000 without knowing to change the other